### PR TITLE
UI-1249 DateInput: use onValueChange

### PR DESCRIPTION
**Pull request by hanzge**
- Remove the original `defaultValue`, `value`, `onChange`. They used to control the display date text of DateInput.
- The user typing event and the display date value is now completely controlled inside the component.
- `name` prop is now `isRequired`
- Change `defaultDateValue` to `defaultValue`
- Change `dateValue` to `value`
- Change `onDateChange` to `onValueChange`
- Update storybook to the new pattern

### CODE REVIEW

File: lib/components/Datepicker/DateInput.js [hunk 1]
```diff
@@ -18,14 +18,10 @@ const propTypes = {
...
   dateInputFormat: PropTypes.string,
-  /**
-   * Input dateValue, follows DEFAULT_DATE_FORMAT
-   */
-  dateValue: PropTypes.string,
   /**
    * default Date value YYYY-MM-DD
    */
-  defaultDateValue: PropTypes.string,
+  defaultValue: PropTypes.string,
```
Comment (joshsylvester) [inline lib/components/Datepicker/DateInput.js]: `defaultValue` isn't needed from the outside - it's only needed in the state of the component for resetting, and can be set from the `value` as the component is rendered.

Comment (hanzge) [reply lib/components/Datepicker/DateInput.js]: @joshsylvester We still need `defaultValue` for initial date value. The `value` is for controlled component usage.

File: lib/components/Datepicker/DateInput.js [hunk 2]
```diff
@@ -74,44 +66,60 @@ const propTypes = {
...
+   */
+  onValueChange: PropTypes.func,
   /**
    * Display the button for jumping to today
    */
   showTodayButton: PropTypes.bool,
+  /**
+   * Input value, follows DEFAULT_DATE_FORMAT
+   */
+  value: PropTypes.string,
```
Comment (joshsylvester) [inline lib/components/Datepicker/DateInput.js]: `value` needs to be removed based on our conversation on 04/11/18.

Comment (hanzge) [reply lib/components/Datepicker/DateInput.js]: `value` is now removed

File: lib/components/Datepicker/DateInput.js [hunk 3]
```diff
@@ -200,33 +192,30 @@ class DateInput extends Component {
...
+      name,
       extension,
       onOpen,
       dateInputFormat,
       showTodayButton,
       ...restProps
     } = this.props;
-    const momentObj = moment(dateValue, DEFAULT_DATE_FORMAT);
-    delete restProps.defaultDateValue;
+    const { value } = this.state;
```
Comment (mallik-svmx) [inline lib/components/Datepicker/DateInput.js]: const { value, inputDisplayValue , isOpen} = this.state;

**Pull request merged**